### DISCUSSION TITLE
Why official images do not use blue-build?

### DISCUSSION THREAD

**Discussion by marcinjahn** [blue-build/.github#26] [General]
I wonder, why don't the images like [bluefin](https://github.com/ublue-os/bluefin) use blue-build, and instead rely on another set of custom scripts, jsons, practices.
Is there any plan to unify it? I think that, especially for new-comers, it's quite confusing.
An example of that is how there are these two templates:

- https://github.com/ublue-os/image-template
- https://github.com/blue-build/template

Both seem to be maintained, what is the intention here?

**Comment by fiftydinar** [comment]
Universal Blue folks prefer pure Containerfile approach, rather than using abstracted approach like BlueBuild.

If they theoretically decided to transition to BlueBuild, it would take a lot of time & testing to migrate to it, since Ublue images, especially Bazzite, use a lot of workarounds for some things.

It's how they want to do things & there's nothing wrong with that.

While we are close to Universal Blue, we are a separate project & we are not directly depending on Universal Blue for decisions that BlueBuild will take. Same goes vice-versa.

So Universal Blue's preference is to use pure Containerfile approach, to make users aware on a more deep level on how Containerfile functions.
Our preference is to use a more abstracted & imo easier approach, while still providing some more in-depth documentation on how Containerfiles work.

So it's all about a choice.

**Reply by marcinjahn** [reply]
Thanks, makes sense. I think it'd be a good idea to mention what you said here, about ublue's way of building images, and why blue-build is different, somewhere on https://blue-build.org/. I guess usually people find out about ublue first, and then some of them stumble upon blue-build. I might be one of many people asking the question "why they're so similar, yet so different".

**Reply by xynydev** [reply]
Did you see the FAQ section? Please let us know if this does not adequately explain things.

https://blue-build.org/#faq

**Reply by marcinjahn** [reply]
Yeah, I did, but still your answer here provides the piece that I was missing from the FAQ: Universal Blue could make use of blue-build, but, for their needs/experimentation, they need more freedom, and they stayed on the level of Containerfile.
